### DISCUSSION TITLE
Make `sign` icon and text configurable?

### DISCUSSION THREAD

**Discussion by yavorski** [linrongbin16/lsp-progress.nvim#130] [Ideas]
Feature request:
Could you make the `sign` icon/text from the defaults final formatter configurable?
```lua
local sign = " LSP"
```
Thank you!

**Comment by linrongbin16** [comment]
hi @yavorski ,

you can overwrite the `format` function in `setup(opts)`.

for example: 

```lua
require('lsp-progress').setup({
  format = function(client_messages)
    ...
  end,
})
```

see: https://github.com/linrongbin16/lsp-progress.nvim/blob/1c37b1cd0611563a767f0a340d61f265c4c54ecd/lua/lsp-progress/defaults.lua#L131

**Comment by yavorski** [comment]
Yeah I know, but i didn't wanted to copy/paste/overwrite the entire function, just to change the icon... Cheers!

**Reply by linrongbin16** [reply]
emmmm, seems no way to do it with a single option now.
